pkg/odo/cli/storage/create.go

```diff
@@ -39,34 +48,77 @@ func NewStorageCreateOptions() *StorageCreateOptions {
 
 // Complete completes StorageCreateOptions after they've been created
 func (o *StorageCreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	o.Context = genericclioptions.NewContext(cmd)
+	if o.isDevfile {
+		o.Context = genericclioptions.NewDevfileContext(cmd)
+
+		o.componentName = o.EnvSpecificInfo.GetName()
+		if o.storageSize == "" {
+			o.storageSize = defaultStorageSize
+		}
+	} else {
+		o.Context = genericclioptions.NewContext(cmd)
+		o.componentName = o.LocalConfigInfo.GetName()
+	}
+
 	if len(args) != 0 {
 		o.storageName = args[0]
 	} else {
-		o.storageName = o.Component() + "-" + util.GenerateRandomString(4)
+		o.storageName = o.componentName + "-" + util.GenerateRandomString(4)
 	}
 	return
 }
 
 // Validate validates the StorageCreateOptions based on completed values
 func (o *StorageCreateOptions) Validate() (err error) {
+	if o.isDevfile {
+		return
+	}
 	// validate storage path
 	return o.LocalConfigInfo.ValidateStorage(o.storageName, o.storagePath)
 }
 
-// Run contains the logic for the odo storage create command
-func (o *StorageCreateOptions) Run() (err error) {
-	storageResult, err := o.LocalConfigInfo.StorageCreate(o.storageName, o.storageSize, o.storagePath)
+func (o *StorageCreateOptions) devFileRun() error {
+	devFile, err := devfile.ParseAndValidate(o.devfilePath)
```

Comment:
Same `s/devFile/devfile`. I think it's called `devfile`. We use `Devfile` only when referring to it in beginning of the sentence. That's what I've observed in the docs PR.

Reply:
devfile collides with a package name here